lib/puppet/type/service.rb

```diff
@@ -60,6 +60,16 @@ module Puppet
         provider.manual_start
       end
 
+      # This only makes sense on systemd systems. Otherwise, it just defaults
+      # to disable.
+      newvalue(:mask, :event => :service_disabled) do
```

Comment:
I'd rather not constrain it. Mask is a weird state that should default to 'disabled' should the system not support masking. We can do it if you'd like, but I think that the way it's set up provides the least amount of Puppet code to implement the expected result.

It's used in quite a few of the built-in types and the code doesn't work properly without it.